warehouse/query-core/src/main/java/datawave/query/iterator/builder/IvaratorBuilder.java

```diff
@@ -44,10 +45,20 @@ protected void validateIvaratorControlDir(IvaratorCacheDir ivaratorCacheDir) {
         String ivaratorCacheDirURI = ivaratorCacheDir.getPathURI();
         FileSystem hdfsFileSystem = ivaratorCacheDir.getFs();
         
-        final URI hdfsCacheURI;
         try {
-            hdfsCacheURI = new URI(ivaratorCacheDirURI);
-            hdfsFileSystem.mkdirs(new Path(hdfsCacheURI));
+            final Path hdfsCachePath = new Path(new URI(ivaratorCacheDirURI));
+            // get the parent directory
+            final Path parentCachePath = hdfsCachePath.getParent();
+            final URI parentURI = parentCachePath.toUri();
+            if (!hdfsFileSystem.exists(parentCachePath)) {
+                // being able to make the parent directory is proof enough
+                hdfsFileSystem.mkdirs(parentCachePath);
+            } else if (parentURI.getScheme().equals("file")) {
```

Comment:
So, if the parent exists in hdfs, we just assume that the directory is valid?  The previous logic would attempt to make a directory whether it's a local filesystem or hdfs - this doesn't.

Reply:
This says hdfsFileSystem but that file systgem could also be a local file system.  I will rename the variable.
That bein said, the parent directory will be created on line 55 regardless.  The subsequent check for the canWrite and canRead can only really be done on the local file system.  The hdfs file system does not support such checks.

Reply:
line 55 won't be hit at all if the parent directory already exists...  that wasn't really my concern.  I was curious why we would perform validation for a local filesystem, but not for hdfs.  Your comments about canWrite and canRead explain that.

However, comparing the new logic with the old, are we doing enough to validate hdfs?  If the parent directory already exists in the hdfs case does that guarantee that we will be able to create the ivarator cache directory?  In the past we tried to make the ivarator cache directory (local or hdfs).  Just wondering if this is equivalent to the previous code, and doing enough to validate for hdfs.

Reply:
OK, I guess I can go back to creating the directory as this one must be created.

Reply:
Ah, now I figured out why I did this.  The problem is that I do not want the directory to be created if the ivarator never actually finds anything (the initial range seek comes up empty).  In that case the directory would be created and then immediately removed which seems wasteful.

Reply:
However creating the directory is finding some holes where I need to cleanup the directory so working through those.

Reply:
Cleanup those holes, but now I have the case where the ivarator never actually runs because an error earlier in the field index lookup.  So I am back to not creating the directory here.